qiskit/transpiler/passes/synthesis/solovay_kitaev_synthesis.py

```diff
@@ -0,0 +1,197 @@
+"""
+A Solovay-Kitaev synthesis plugin to Qiskit's transpiler.
+"""
+
+from __future__ import annotations
+
+import numpy as np
+
+from qiskit.converters import circuit_to_dag
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.synthesis.discrete_basis.solovay_kitaev import SolovayKitaevDecomposition
+from qiskit.synthesis.discrete_basis.generate_basis_approximations import (
+    generate_basic_approximations,
+)
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.exceptions import TranspilerError
+
+from .plugin import UnitarySynthesisPlugin
+
+
+class SolovayKitaev(TransformationPass):
+    r"""Approximately decompose 1q gates to a discrete basis using the Solovay-Kitaev algorithm.
+
+    See :mod:`qiskit.synthesis.discrete_basis` for more information.
+    """
+
+    def __init__(
+        self,
+        recursion_degree: int = 3,
+        basic_approximations: str | dict[str, np.ndarray] | None = None,
+    ) -> None:
+        """
+        Args:
+            recursion_degree: The recursion depth for the Solovay-Kitaev algorithm.
+                A larger recursion depth increases the accuracy and length of the
+                decomposition.
+            basic_approximations: The basic approximations for the finding the best discrete
+                decomposition at the root of the recursion. If a string, it specifies the ``.npy``
+                file to load the approximations from. If a dictionary, it contains
+                ``{label: SO(3)-matrix}`` pairs. If None, a default based on the H, T and Tdg gates
+                up to combinations of depth 10 is generated.
+        """
+        super().__init__()
+        self.recursion_degree = recursion_degree
+        self._sk = SolovayKitaevDecomposition(basic_approximations)
+
+    def run(self, dag: DAGCircuit) -> DAGCircuit:
+        """Run the ``SolovayKitaev`` pass on `dag`.
+
+        Args:
+            dag: The input dag.
+
+        Returns:
+            Output dag with 1q gates synthesized in the discrete target basis.
+
+        Raises:
+            TranspilerError: if a gates does not have to_matrix
+        """
+        for node in dag.op_nodes():
+            if not node.op.num_qubits == 1:
+                continue  # ignore all non-single qubit gates
+
+            if not hasattr(node.op, "to_matrix"):
+                raise TranspilerError(
+                    "SolovayKitaev does not support gate without "
+                    f"to_matrix method: {node.op.name}"
+                )
+
+            matrix = node.op.to_matrix()
+
+            # call solovay kitaev
+            approximation = self._sk.run(matrix, self.recursion_degree)
+
+            # convert to a dag and replace the gate by the approximation
+            substitute = circuit_to_dag(approximation)
+            dag.substitute_node_with_dag(node, substitute)
+
+        return dag
+
+
+class SolovayKitaevSynthesis(UnitarySynthesisPlugin):
```

Comment:
I feel like these docs should be exposed somewhere probably from the qiskit (probably off of: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/transpiler/passes/synthesis/plugin.py#L239 ) so we can document the configuration options.

Reply:
Hmm it feels a little strange to change `plugin.py` even though there's nothing SK specific there. Or did you mean adding another `rst` for `qiskit.transpiler.passes.synthesis`?